react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md

```diff
@@ -0,0 +1,188 @@
+
+~~~jsx
+function Button() {
+  return (
+    <button>Click Me!</button>
+  )
+}
+
+export default function App() {
+  return (
+    <div>
+      <Button />
+      <Button />
+      <Button />
+    </div>
+  )
+}
+~~~
+So far so good right? We have a beautiful reusable button that we can use as many times as we like, there is just one small problem. 
+
+What if we wanted the text within our second button to be “Don’t Click Me!’? Right now, we would have to create a second button component with this different text.
+
+~~~jsx
+function Button() {
+  return (
+    <button>Click Me!</button>
+  )
+}
+
+function Button2() {
+  return (
+    <button>Don't Click Me!</button>
+  )
+}
+
+export default function App() {
+  return (
+    <div>
+      <Button />
+      <Button2 />
+      <Button />
+    </div>
+  )
+}
+~~~
+This may not seem like a huge deal right now, but what if we had 10 buttons, each one having different text, fonts, colors, sizes, and any other variation you can think of. Creating a new component for each of these button variations would very quickly lead to a LOT of code duplication. 
+
+Let's see how by using props, we can account for any number of variations with a *single* button component.
+
+~~~jsx
+function Button(props) {
+
+  const buttonStyle = {
+    color: props.color,
+    fontSize: props.fontSize + 'px'
+  };
+
+  return (
+    <button style={buttonStyle}>{props.text}</button>
+  )
+}
+
+export default function App() {
+  return (
+    <div>
+      <Button text="Click Me!" color="blue" fontSize={12} />
+      <Button text="Don't Click Me!" color="red" fontSize={12} />
+      <Button text="Click Me!" color="blue" fontSize={20} />
+    </div>
+  );
+}
+~~~
+There are a few things going on here.
+
+- The `Button` functional component now receives `props` as a function argument. The individual properties are then referenced within the component via `props.propertyName`.
+- When rendering the `Button` components within `App`, the `prop` values are defined on each component.
+- Inline styles are dynamically generated and then applied to the `button` element.
+
+### Prop Destructuring
+
+A very common pattern you will come across in React is prop [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment). Unpacking your props in the component arguments allows for more concise and readable code. Check out prop destructing in action in the example below.
+
+~~~jsx
+function Button({ text, color, fontSize }) {
+  const buttonStyle = {
+    color: color,
+    fontSize: fontSize + "px"
+  };
+
+  return <button style={buttonStyle}>{text}</button>;
+}
+
+export default function App() {
+  return (
+    <div>
+      <Button text="Click Me!" color="blue" fontSize={12} />
+      <Button text="Don't Click Me!" color="red" fontSize={12} />
+      <Button text="Click Me!" color="blue" fontSize={20} />
+    </div>
+  );
+}
+~~~
+### Default Props
```

Comment:
Can we add a linebreak before the Default Props heading? Our markdown parser wants a linebreak here🤷 

![image](https://user-images.githubusercontent.com/85733202/198866961-1d981259-4273-4099-be8b-769c3056e47b.png)
```suggestion
~~~

### Default Props
```